### PR TITLE
Make aiopg installable in PyPy environment

### PR DESCRIPTION
**Pull request by avanov**

### CODE REVIEW

**Comment by asvetlov** [comment]
Please enable PyPy in travis config to make sure that library is really functional on PyPy.

**Comment by avanov** [comment]
Is there a list of pypy versions available on Travis CI, or do they follow the same naming scheme as pyenv does? I test the package against `pypy3.5-5.8.0` locally.

**Comment by asvetlov** [comment]
https://docs.travis-ci.com/user/languages/python/#PyPy-Support

**Comment by avanov** [comment]
Ok, currently builds fail on PyPy because of the presence of `psycopg2` in `requirements.txt`. What is the reason for having psycopg2 pinned in `requirements.txt`?

File: aiopg/utils.py [hunk 1]
```diff
@@ -231,3 +233,10 @@ def __exit__(self, *args):
         coroutines._COROUTINE_TYPES += (_ContextManager,)
     except:
         pass
+
+
+# map psycopg2cffi to psycopg2, so any code
+# that imports psycopg2 will use psycopg2cffi
+if PY_IMPL == 'PyPy':
+    from psycopg2cffi import compat
+    compat.register()
```
Comment (jettify) [inline aiopg/utils.py]: I have never used `psycopg2cffi`, but what happened if `compat.register()` called two times? or after first import of `psycopg2` in other module?

Comment (avanov) [inline aiopg/utils.py]: @jettify it import `psycopg2cffi` and makes it available under `psycopg2` name in `sys.modules` (see https://github.com/svpcom/psycopg2cffi/blob/master/psycopg2cffi/compat.py)

As long as`aiopg.utils` is imported in the main package namespace before `psycopg2`, all subsequent occurences of `psycopg2` will be mapped to already imported `psycopg2cffi` (including relative import cases that go through the main `aiopg` namespace anyway).

Comment (asvetlov) [inline aiopg/utils.py]: Hmm. 
Thanks @jettify for pointing on.
Maybe better to explicitly import either `psycopg2` or `psycopg2cffi`?
Doing this we'll never catch an import order problem.

Comment (avanov) [inline aiopg/utils.py]: @asvetlov do you mean you would prefer `psycopg2_compat` module with sth like this inside?

```python
# aiopg/psycopg2_compat

if PY_IMPL == 'PyPy':
    from psycopg2cffi import *
else:
    from psycopg2 import *
```

Comment (asvetlov) [inline aiopg/utils.py]: Yes,

**Comment by avanov** [comment]
@asvetlov shall I remove `psycopg2` from `requirements.txt` ?

**Comment by jettify** [comment]
We use `requirements.txt` to track updated with pyup bot, so can be removed, I believe.

**Comment by avanov** [comment]
The test failures for CPython environments are now consistent with https://github.com/aio-libs/aiopg/pull/362 due to changes introduced to `psycopg2` between version `2.6.2` and `2.7.3`. TravisCI now fetches the most recent psycopg2 version during the `python setup.py install` build step.

**Comment by steven-upside** [comment]
I see that this commit has been approved, are there plans to merge this anytime soon? PyPy + aiopg would be a great combination!

**Comment by asvetlov** [comment]
1. PR was approved by arbitrary github user, not aiopg committer. Basically it is an another form for saying "I like it, +1"
2. Tests are still fails, the Pull Request is not ready for merging yet.

**Comment by avanov** [comment]
@steven-upside so yes, I am waiting for a response from psycopg2cffi on the issue above, and on another one at https://github.com/chtd/psycopg2cffi/issues/85 , after which I think I could finalise this PR.

**Comment by avanov** [comment]
closing this as no longer relevant

**Pull request closed without merging**